### DISCUSSION TITLE
Create self box with Packer.

### DISCUSSION THREAD

**Discussion by tinntvs** [gusztavvargadr/packer#380] [Q&A] [answered]
I am a newbie from Vietnam and I would like to create a personal Vagrant box and upload it to the Vagrant Cloud for use with Vagrant.
I am currently looking for guidelines on how to use Packer to build a box from an ISO. However, I am facing problems with the Boot_Command in the Packer JSON file.
I noticed that you have created several boxes on Vagrant Cloud and I was wondering if you could share your JSON files for creating them, such as Windows Server, Windows 10 22H2, Ubuntu 18.04, 20.04, and more.
It would be greatly appreciated if you could share them with everyone. Thank you so much!

**Comment by gusztavvargadr** [comment]
@tinntvs thanks for your question.

Please take a look at the [documentation in this repo](https://github.com/gusztavvargadr/packer#usage), the whole process of building a box - including generating the Packer templates - is described there, all the sources are public, so it is already shared with everyone.

You can find the individual parts for each OS and component under the usual [src folder](https://github.com/gusztavvargadr/packer/tree/master/src).

I hope this helps, please let me know.

**Reply by tinntvs** [reply]
I am sorry! I found the individual parts for each OS under the 'src' folder, but I cannot see the templates for each OS. It appears that I was not careful enough to read the README. Thank you so much!

**Reply by gusztavvargadr** [reply, marked as answer]
No worries, this approach I use is in this repo is a bit unorthodox, indeed, the full templates are constructed during the build, as the first step, with the `--target=restore` switch. After you run this restore stage, the resulting folders under the `build` folder should be working in a standalone setup as well, so you can check out the full templates from there.
